lib/helper/WebDriver.js

```diff
@@ -640,6 +697,19 @@ class WebDriver extends Helper {
   async _locate(locator, smartWait = false) {
     if (require('../store').debugMode) smartWait = false;
 
+    // special locator type for Shadow DOM
+    if (this._isShadowLocator(locator)) {
+      if (!this.options.smartWait || !smartWait) {
+        const els = await this._locateShadow(locator);
+        return els;
+      }
+
+      await this._smartWait(locator);
```

Comment:
Smart wait won't work with shadow dom
```suggestion
```

Reply:
IWhen I try it waits and also incorporates the `retry`. Pls see below, 

```
I click {"shadow":["downloads-manager","downloads-toolbar","cr-toolbar","cr-toolbar-search-field","[title=\"Search downloads\"]"]}
    › [SmartWait (5000ms)] Locating {shadow: downloads-manager,downloads-toolbar,cr-toolbar,cr-toolbar-search-field,[title="Search downloads"]} in 5000
    [1] Retrying... Attempt #2
    › [SmartWait (5000ms)] Locating {shadow: downloads-manager,downloads-toolbar,cr-toolbar,cr-toolbar-search-field,[title="Search downloads"]} in 5000
    [1] Retrying... Attempt #3
    › [SmartWait (5000ms)] Locating {shadow: downloads-manager,downloads-toolbar,cr-toolbar,cr-toolbar-search-field,[title="Search downloads"]} in 5000
    [1] Retrying... Attempt #4
    › [SmartWait (5000ms)] Locating {shadow: downloads-manager,downloads-toolbar,cr-toolbar,cr-toolbar-search-field,[title="Search downloads"]} in 5000
    [1] Retrying... Attempt #5
    › [SmartWait (5000ms)] Locating {shadow: downloads-manager,downloads-toolbar,cr-toolbar,cr-toolbar-search-field,[title="Search downloads"]} in 5000
```